pandas/tests/io/formats/test_format.py

```diff
@@ -3224,6 +3234,22 @@ def test_period_tz(self):
             per = dt.to_period(freq="H")
         assert per.format()[0] == "2013-01-01 00:00"
 
+    def test_period_custom_locale(self, overridden_locale):
+        # GH#46319 locale-specific formatting directive leads to non-utf8 str result
+
+        # Get locale-specific reference
+        am_local, pm_local = get_local_am_pm()
+
+        # Scalar
+        per = pd.Period("2018-03-11 13:00", freq="H")
+        assert per.strftime("%p") == pm_local
+
+        # Index
+        per = pd.period_range("2003-01-01 01:00:00", periods=2, freq="12h")
+        formatted = per.format(date_format="%y %I:%M:%S%p")
```

Comment:
would throwing some non-ascii characters in there be useful?  or just a distraction?

Reply:
good idea ! I'll have a try

Reply:
It indeed made thetest fail :) so I created an issue and fixed it

Reply:
I eventually created two separate tests

Reply:
I went for the latter.